rcsbapi/data/schema.py

```diff
@@ -437,13 +450,21 @@ def __construct_query_rustworkx(self, input_ids, input_type, return_data_list):
                 break
 
         target_node_indices = []
-        for return_data in return_data_list:
-            for node in self.schema_graph.node_indices():
-                node_data = self.schema_graph[node]
-                if isinstance(node_data, FieldNode) and node_data.name == return_data:
-                    target_node_indices.append(node_data.index)
-                    break
-        # print(target_node_indices)
+        if all('.' not in rd for rd in return_data_list):
```

Comment:
I think these cases could be combined. The node index should have the name that is passed in. So in the non-dot notation case, 'node_index_dict[field_name]' should return the index. In the dot notation case, 'node_index_dict[parent_type.field_name]' should return the index. I don't think you need to iterate over the node indices.